### PR TITLE
Refactor/finesse UI

### LINKED ISSUE #49: Finesse UI
Products component:
- [x] height of section
- [x] text decoration
- [x] product titles - add a space between 2-word titles (use helper function?)

General
- [x] background color on product detail page
- [x] fonts - use Open Sans, different font weights for headers vs body text
- [x] create a mixin for consistent button styling
- [x] box border mixin

### PR DESCRIPTION
**Pull request by rachelsoae**
What I did:
- Created uniform button across site (except Landing header)
- Ensured uniform color and border radius scheme across Product Detail, Shopping Bag, and Checkout pages
- Fixed formatting of color names on Checkout page
- Changed Nav arrow from png to icon
- Refactored most style pages to reduce nesting for readability and maximize utility of mixins/variables
- Added box shadows to Shopping Bag items
Closes #49 

Did this break anything?
- [x] No
- [ ] Yes

Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

Checklist:
- [x] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [x] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

What's next?

### CODE REVIEW

File: src/styles/_ProductDetail.scss [hunk 1]
```diff
@@ -6,14 +6,17 @@
...
     background-image: url('../images/detail-horizontal-bg.jpg');
     background-size: cover;
     height: 15rem;
-    display: flex;
-    align-items: center;
-    padding: 3rem;
+    width: 100%;
+    @include m.horizontally-aligned-flex;
+    justify-content: center;
+    padding: 3rem 0;
```
Comment (taranveersingh93) [inline src/styles/_ProductDetail.scss:19]: I think we should have some padding on the side too. It looks weird touching the sides.

File: src/styles/_ProductDetailOrderForm.scss [hunk 2]
```diff
@@ -28,22 +31,22 @@
...
+      @include m.box-border;
       height: 3rem;
       font-size: 1.5rem;
-      color: v.$primary-accent-color;
+      @include m.body-font;
+    }
+
+    select:hover {
+      cursor: pointer;
+      border: 2px solid v.$primary-accent-color;
```
Comment (taranveersingh93) [inline src/styles/_ProductDetailOrderForm.scss:49]: The hover state doesn't work with phone screens. If you interact with the form on the phone screen sized window, clicks would register as a hover, resulting in persistent hover styling. I fix this by having hover styling in a `min-width: 800px)` media query (so it's activated only on wider screens).

File: src/styles/_ProductDetailOrderForm.scss [hunk 3]
```diff
@@ -98,17 +101,9 @@
...
-@media (min-width: 800px) {
-  .details-order-form__selection-pair {
-    select:hover {
-      cursor: pointer;
-      border: 2px solid v.$primary-accent-color;
+    .details-order-form__hidden {
+      display: none;
     }
   }
-  .details-order-form__submit-btn:hover {
```
Comment (taranveersingh93) [inline src/styles/_ProductDetailOrderForm.scss:111]: The hover should stay in the wide screen media query.

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]
Looks good. Requested some breaking changes that are crucial to fix.

File: src/styles/_ProductDetailOrderForm.scss (hunk 3 above)
Comment (rachelsoae) [reply src/styles/_ProductDetailOrderForm.scss:111]: I respectfully disagree. If I have the window open on my computer screen but I have it minimized to only take up a fraction of the screen, the hover states should still work. Having hovers does not hurt anything in mobile view.

**Review by alivaditis — COMMENTED** [review]
Great job Rachel, it looks very professional!

One thing I'm catching just now is that the data is still being passed in camel so we see something like this 
Screen Shot 2023-09-03 at 7 14 47 PM

on line 51-53 of Form we would have to use the camelToPascal and maybe a toFixed(2)?

`        return {
          name: item.type,
          color: item.color,
          size: item.size,
          quantity: parseInt(item.quantity)
        }
`
if it's not something you're able to change rn I can make a ticket for it

File: src/styles/_ProductDetail.scss (hunk 1 above)
Comment (rachelsoae) [reply src/styles/_ProductDetail.scss:19]: Also respectfully disagree with this. The photo is a banner and I think it looks extremely odd if it is not touching the sides.

**Comment by rachelsoae** [comment]
> Great job Rachel, it looks very professional!
> 
> One thing I'm catching just now is that the data is still being passed in camel so we see something like this Screen Shot 2023-09-03 at 7 14 47 PM
> 
> on line 51-53 of Form we would have to use the camelToPascal and maybe a toFixed(2)?
> 
> ` return { name: item.type, color: item.color, size: item.size, quantity: parseInt(item.quantity) }` if it's not something you're able to change rn I can make a ticket for it

I think this should definitely be changed and should be popped out as a new issue.

File: src/styles/_ProductDetail.scss (hunk 1 above)
Comment (alivaditis) [reply src/styles/_ProductDetail.scss:19]: I am with Rachel on this one, sorry Taranveer

Comment (taranveersingh93) [reply src/styles/_ProductDetail.scss:19]: To clarify: The banner picture should touch the sides. IMO, The text shouldn't touch the border of the picture. Padding would affect the text's distance from the picture's border. Not the picture's border from the screen width. 

Do we still have opposing views? Usually flexible on styling but text touching the sides seems really icky.

Comment (JamieCaudill) [reply src/styles/_ProductDetail.scss:19]: I think that padding is a good idea because there is padding everywhere else.

**Review by alivaditis — APPROVED** [review]
It looks so good!

File: src/styles/_ProductDetailOrderForm.scss (hunk 3 above)
Comment (taranveersingh93) [reply src/styles/_ProductDetailOrderForm.scss:111]: Definitely need a tie breaker on this. 800 px is REALLY narrow. If a user misses hover states on that narrow of a screen, I think it's ok. It's not even your laptop screen width.

Whereas, having hover states on all screens would definitely contribute to a poor experience for the mobile user who'd see hover state, and then active state. 

With the hover state active on mobile screen, it looks like this:
Screenshot 2023-09-03 at 7 32 46 PM
when it should look like this
Screenshot 2023-09-03 at 7 32 46 PM

File: src/styles/_Landing.scss [hunk 4]
```diff
@@ -10,42 +10,40 @@
...
+  @include m.true-center;
+  flex-direction: column;
 
   .landing__header {
     height: 25rem;
-    display: flex;
     margin: 0;
-    flex-direction: column;
-    justify-content: center;
+    display: flex;
```
Comment (JamieCaudill) [inline src/styles/_Landing.scss:19]: Redundant

File: src/styles/_ProductDetail.scss (hunk 1 above)
Comment (taranveersingh93) [reply src/styles/_ProductDetail.scss:19]: Screenshot 2023-09-03 at 7 38 53 PM
I'm talking about this. The text touching the sides of the picture.

**Review by JamieCaudill — APPROVED** [review]
Looks good! That arrow that closes the right nav looks great! Nice work.

File: src/styles/_Landing.scss (hunk 4 above)
Comment (rachelsoae) [reply src/styles/_Landing.scss:19]: Good catch! Deleted. Thanks.

File: src/styles/_ProductDetailOrderForm.scss (hunk 3 above)
Comment (alivaditis) [reply src/styles/_ProductDetailOrderForm.scss:111]: I don't know what I'm looking at tbh, is it the color should be darker but it is always stuck in the hover color?

File: src/styles/_ProductDetail.scss (hunk 1 above)
Comment (rachelsoae) [reply src/styles/_ProductDetail.scss:19]: Per our conversation, added max-width: 85% to banner text - looks good at all screen sizes. Thanks for noticing this!

File: src/styles/_ProductDetailOrderForm.scss (hunk 2 above)
Comment (rachelsoae) [reply src/styles/_ProductDetailOrderForm.scss:49]: Per our conversation, you will fix in a separate issue. Thank you for explaining this to me :) I agree with you!

**Comment by rachelsoae** [comment]
> Looks good! That arrow that closes the right nav looks great! Nice work.

Yay, glad you like it!!

**Review by taranveersingh93 — APPROVED** [review]
Thank you for taking this on and for adding the padding to text. Merging. The other item can be a separate issue.

**Pull request merged**